bubblewrap.c

```diff
@@ -609,7 +617,8 @@ setup_newroot (bool unshare_pid,
             die_with_error ("Can't get type of source %s", op->source);
         }
 
-      if (op->dest)
+      if (op->dest &&
+          op->type != SETUP_SET_HOSTNAME)
```

Comment:
This is a little too magic. Can you add a SetupOp "bool no_create_dest" member and set that instead?